.changeset/perfect-horses-tell.md

```diff
@@ -0,0 +1,27 @@
+---
+'astro': major
+---
+
+Astro's JSX handling has been refactored with better support for each framework.
+
+Previously, Astro automatically scanned your components to determine which framework-specific transformations should be used. In practice, supporting advanced features like Fast Refresh with this approach proved difficult.
+
+Now, rather than scanning your components, JSX framework integrations accept `include` and `exclude` config options. When using multiple JSX frameworks in the same project, users should manually control which files belong to each framework using the `include` and `exclude` options.
```

Comment:
```suggestion
Now, Astro determines which framework to use with `include` and `exclude` config options where you can specify files and folders on a per-framework basis. When using multiple JSX frameworks in the same project, users should manually control which files belong to each framework using the `include` and `exclude` options.
```
Is this correct? I feel like it's helpful to call out these config values. 

Also, Is this just an array? Can there be more than one item in an `include` or `exclude`? Can it be an individual file instead of a path? Can you BOTH include and exclude things for a framework? I think having one of these be a more complex example will be helpful at least by the time we get to the migration guide, so I'll want to do that in the Docs Issue, but I'll let you decide whether that's overloading this changeset!

Reply:
Yes, there can be any number of items in the include/exclude. Yes, you can add both. I didn't add an `exclude` example as I don't know a great use-case for it. But it's something that's provided by each framework plugin so we are just exposing that to the user.